lib/matplotlib/legend_handler.py

```diff
@@ -609,23 +612,43 @@ def create_artists(self, legend, orig_handle,
         leg_stemlines = [Line2D([x, x], [bottom, y])
                          for x, y in zip(xdata_marker, ydata)]
 
-        for lm, m in zip(leg_stemlines, stemlines):
-            self.update_prop(lm, m, legend)
+        if using_linecoll:
+            # update_prop() usually takes two Line2D collections;
+            # override temporarily to copy properties from a LineCollection
```

Comment:
Found this confusing, and had to look elsewhere in code for explanation.  Suggest: 

"change the function used by update_prop() from the default to one that handles LineCollection"